wicker/core/storage.py

```diff
@@ -215,64 +215,115 @@ class S3PathFactory:
                     - part-1-attempt-2345.parquet
     """
 
-    def __init__(self, s3_root_path: str = None) -> None:
-        s3_config = get_config().aws_s3_config
-        if s3_root_path is None:
-            s3_root_path = s3_config.s3_datasets_path
-        self.root_path = s3_root_path
-        self.store_concatenated_bytes_files_in_dataset = s3_config.store_concatenated_bytes_files_in_dataset
+    def __init__(self, root_path: str, store_concatenated_bytes_files_in_dataset: bool = False) -> None:
+        """Init the path factory.
+
+        Object to form the expected paths and return them to the user based of root path and storage bool.
+
+        Args:.
+            root_path (str): File system loc of the root of the wicker file structure.
+            store_concatenated_bytes_files_in_dataset (bool, optional): Whether to assume concat bytes files are stored.
+                Defaults to False
+        """
+        self.root_path: str = root_path
+        self.store_concatenated_bytes_files_in_dataset = store_concatenated_bytes_files_in_dataset
 
     def __eq__(self, other: Any) -> bool:
         return super().__eq__(other) and type(self) == type(other) and self.root_path == other.root_path
 
-    def get_dataset_assets_path(self, dataset_id: DatasetID, s3_prefix: bool = True) -> str:
+    def _get_dataset_assets_path(self, dataset_id: DatasetID, prefix: Optional[str] = None) -> str:
```

Comment:
The naming `prefix` here suggests that it is about to be added rather remove, should we rename to something like `prefix_to_drop`? Similar comment applies elsewhere

Reply:
How about I do it in a later PR? The naming is not good on the s3_prefix param either and I want to rename them all at once in a different PR.

Reply:
Since we have just introduced the `prefix` name in this PR, and it is minimal effort to change that also in this PR, I suggest to tackle it here and not introduce a new debt. 

The `s3_prefix` is external to this repo and I agree that we might need to do more work to change it elsewhere and it is better to be a separate PR.

Reply:
@isaak-willett I also got a little bit confused, maybe we should call it `prefix_remove` or `prefix_trim`.

Reply:
Changed to `prefix_to_trim`

Reply:
Changed to `prefix_to_trim`